Automattic/jetpack
# Protect: Add fixer status to initial state

## Description
Retrieve status of all fixable threats in the initial state

### Other information:

## Jetpack product discussion


- https://github.com/Automattic/jetpack-scan-team/issues/1389

## Does this pull request change what data or activity we track or use?


- No

## Testing instructions:






* Checkout this branch and start Jurassic Tube
* From the browser dev console log out `jetpackProtectInitialState` and verify that: 
   * `status.fixable_threat_ids` is set to the default empty array
   * `fixerStatus` is `false` as an upgrade is required to retrieve the fixer status data
* Activate Protect and upgrade
* Add fixable threats using the Debug Tool
* Run the scan to identify the threats
* From the browser dev console log out `jetpackProtectInitialState` and verify that: 
   * `status.fixable_threat_ids` is populated with the IDs for the fixable threats that scan identified
   * `fixerStatus` is still `false` as a user connection is required to retrieve the fixer status data
* Connect your user account
* From the browser dev console log out `jetpackProtectInitialState` and verify that:
   * `status.fixable_threat_ids` is populated with the IDs for the fixable threats that scan identified
   * `fixerStatus` is populated with status of the IDs
